### PR TITLE
[enhancement](Nereids) reduce CostAndEnforcerJob call times

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

record pruned plan's cost to avoid optimize same GroupExpression more than once.

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java [hunk 1]
```diff
@@ -139,7 +165,28 @@ public List<GroupExpression> clearPhysicalExpressions() {
     }
 
     public double getCostLowerBound() {
-        return costLowerBound;
+        return -1D;
+    }
+
+    /**
+     * Get the lowest cost {@link org.apache.doris.nereids.trees.plans.physical.PhysicalPlan}
+     * which meeting the physical property constraints in this Group.
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java:173]: meeting -> satisfies

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java [hunk 2]
```diff
@@ -257,6 +257,65 @@ public int removeParentExpression(GroupExpression parent) {
         return parentExpressions.size();
     }
 
+    /**
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java:260]: why move the position of these functions?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java [hunk 3]
```diff
@@ -139,7 +165,28 @@ public List<GroupExpression> clearPhysicalExpressions() {
     }
 
     public double getCostLowerBound() {
-        return costLowerBound;
+        return -1D;
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java:168]: I think -0.01 is not an acceptable bound. who about return 0.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java (hunk 1 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java:173]: this is original comment, i do not mod it at all. all comment should re-check together

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java (hunk 2 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java:260]: put correlative function together for easy reading

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java (hunk 3 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Group.java:168]: this is original value, just remove useless variables

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.28 seconds
 load time: 435 seconds
 storage size: 17173193941 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221123084513_clickbench_pr_51445.html

**Review by jackwener — APPROVED** [review]

**Pull request merged**